### PR TITLE
Wip/hughsie/break plugin api

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: contrib/fwupd.spec.in [hunk 1]
```diff
@@ -5,6 +5,7 @@
 %global libjcat_version 0.1.0
 %global systemd_version 231
 %global json_glib_version 1.1.1
+%global libfwupdplugin_lt_current 3
```
Comment (superm1) [inline contrib/fwupd.spec.in:8]: changes to this file should probably be a separate commit

File: contrib/fwupd.spec.in [hunk 2]
```diff
@@ -401,99 +402,99 @@ done
 %{_libdir}/girepository-1.0/FwupdPlugin-1.0.typelib
 /usr/lib/udev/rules.d/*.rules
 /usr/lib/systemd/system-shutdown/fwupd.shutdown
-%dir %{_libdir}/fwupd-plugins-3
```
Comment (superm1) [inline contrib/fwupd.spec.in:404]: I guess we made a mistake at some point with this directory naming?

File: libfwupdplugin/fu-plugin.c [hunk 3]
```diff
@@ -1159,9 +1159,11 @@ gboolean
 fu_plugin_runner_update_prepare (FuPlugin *self, FwupdInstallFlags flags, FuDevice *device,
```
Comment (superm1) [inline libfwupdplugin/fu-plugin.c:1159]: maybe these runner functions should rename too

File: contrib/fwupd.spec.in (hunk 1 above)
Comment (superm1) [reply contrib/fwupd.spec.in:8]: Also, since this is "spec.in", can't this come from meson.build in a `configure()`?

Comment (hughsie) [reply contrib/fwupd.spec.in:8]: I guess, but I think it changes so little it's not worth the build pain.

File: contrib/fwupd.spec.in (hunk 2 above)
Comment (hughsie) [reply contrib/fwupd.spec.in:404]: Maybe; I think originally the soname of libfwupdplugin and and the plugin ABI were different, but realistically they're coupled and I think it makes sense to keep them the same. Maybe it makes sense to jump to **4** and make sure both change?

Comment (superm1) [reply contrib/fwupd.spec.in:404]: > Maybe; I think originally the soname of libfwupdplugin and and the plugin ABI were different, but realistically they're coupled and I think it makes sense to keep them the same. Maybe it makes sense to jump to **4** and make sure both change?

Yes

File: contrib/fwupd.spec.in (hunk 1 above)
Comment (hughsie) [reply contrib/fwupd.spec.in:8]: Actually, it's just 4 lines of pain. Changed, good idea.

File: contrib/fwupd.spec.in (hunk 2 above)
Comment (hughsie) [reply contrib/fwupd.spec.in:404]: I've moved everything to 4 now.

File: libfwupdplugin/fu-plugin.c (hunk 3 above)
Comment (hughsie) [reply libfwupdplugin/fu-plugin.c:1159]: Good idea, fixed.

File: contrib/debian/lintian/fwupd [hunk 4]
```diff
@@ -3,13 +3,13 @@ fwupd binary: systemd-service-file-missing-install-key lib/systemd/system/fwupd-
...
-fwupd: library-not-linked-against-libc usr/lib/*/fwupd-plugins-3/libfu_plugin_modem_manager.so
-fwupd: library-not-linked-against-libc usr/lib/*/fwupd-plugins-3/libfu_plugin_pci_bcr.so
-fwupd: library-not-linked-against-libc usr/lib/*/fwupd-plugins-3/libfu_plugin_pci_mei.so
-fwupd: library-not-linked-against-libc usr/lib/*/fwupd-plugins-3/libfu_plugin_iommu.so
-fwupd: library-not-linked-against-libc usr/lib/*/fwupd-plugins-3/libfu_plugin_msr.so
+fwupd: library-not-linked-against-libc usr/lib/*/fwupd-plugins-4/libfu_plugin_modem_manager.so
+fwupd: library-not-linked-against-libc usr/lib/*/fwupd-plugins-4/libfu_plugin_pci_bcr.so
+fwupd: library-not-linked-against-libc usr/lib/*/fwupd-plugins-4/libfu_plugin_pci_mei.so
+fwupd: library-not-linked-against-libc usr/lib/*/fwupd-plugins-4/libfu_plugin_iommu.so
+fwupd: library-not-linked-against-libc usr/lib/*/fwupd-plugins-4/libfu_plugin_msr.so
```
Comment (superm1) [inline contrib/debian/lintian/fwupd:15]: so as to not have to touch this again, maybe it should be `fwupd-plugins-*`

File: contrib/debian/fwupd-tests.install [hunk 5]
```diff
@@ -5,7 +5,7 @@
 usr/share/installed-tests/*
 usr/libexec/installed-tests/fwupd/fwupd.sh
 usr/libexec/installed-tests/fwupd/*-self-test
-usr/lib/*/fwupd-plugins-3/libfu_plugin_test.so
-usr/lib/*/fwupd-plugins-3/libfu_plugin_invalid.so
+usr/lib/*/fwupd-plugins-4/libfu_plugin_test.so
+usr/lib/*/fwupd-plugins-4/libfu_plugin_invalid.so
```
Comment (superm1) [inline contrib/debian/fwupd-tests.install:9]: same here, maybe make it `fwupd-plugins-*`

File: contrib/debian/rules [hunk 6]
```diff
@@ -66,9 +66,9 @@ override_dh_install:
 	dh_missing -a --fail-missing
 
 	#this is placed in fwupd-tests
-	rm -f debian/fwupd/usr/lib/*/fwupd-plugins-3/libfu_plugin_test.so
-	rm -f debian/fwupd/usr/lib/*/fwupd-plugins-3/libfu_plugin_test_ble.so
-	rm -f debian/fwupd/usr/lib/*/fwupd-plugins-3/libfu_plugin_invalid.so
+	rm -f debian/fwupd/usr/lib/*/fwupd-plugins-4/libfu_plugin_test.so
+	rm -f debian/fwupd/usr/lib/*/fwupd-plugins-4/libfu_plugin_test_ble.so
+	rm -f debian/fwupd/usr/lib/*/fwupd-plugins-4/libfu_plugin_invalid.so
```
Comment (superm1) [inline contrib/debian/rules:71]: `fwupd-plugins-*`

File: contrib/meson.build [hunk 7]
```diff
@@ -3,10 +3,17 @@ if get_option('qubes')
...
-if host_machine.system() == 'windows'
-  con2 = configuration_data()
-  con2.set('FWUPD_VERSION', fwupd_version)
+con2 = configuration_data()
+con2.set('FWUPD_VERSION', fwupd_version)
+con2.set('FWUPD_PLUGINVER', libfwupdplugin_lt_current)
+
+configure_file(
+  input : 'fwupd.spec.in',
+  output : 'fwupd.spec.in',
```
Comment (superm1) [inline contrib/meson.build:12]: shouldn't the output be `fwupd.spec?`

File: contrib/setup-win32.nsi [hunk 8]
```diff
@@ -64,13 +64,13 @@ Section "fwupd"
...
-  File "libfwupdplugin-2.dll"
+  File "libfwupdplugin-4.dll"
   File "libgcab-1.0-0.dll"
   File "libxmlb-2.dll"
   File "libjcat-1.dll"
   File "libgusb-2.dll"
-  SetOutPath "$INSTDIR\fwupd-plugins-3"
-  File /r "fwupd-plugins-3/libfu_plugin_*.dll"
+  SetOutPath "$INSTDIR\fwupd-plugins-4"
+  File /r "fwupd-plugins-4/libfu_plugin_*.dll"
```
Comment (superm1) [inline contrib/setup-win32.nsi:73]: `fwupd-plugins-*`?

**Review by superm1 — APPROVED** [review]
I pushed up the things that I thought made sense, LGTM now.

File: contrib/meson.build (hunk 7 above)
Comment (hughsie) [reply contrib/meson.build:12]: Actually .in -- it's preprocessed again by a different tool... :)
